code/messaging/src/phone/java/com/adobe/marketing/mobile/Messaging.java

```diff
@@ -305,13 +307,17 @@ public void call(final Event event) {
                         return;
                     }
 
-                    for (final Surface surface : surfaces) {
-                        for (final Map<String, Object> propositionMap : retrievedPropositions) {
-                            final Proposition proposition = Proposition.fromEventData(propositionMap);
-                            propositions.add(proposition);
-                        }
+                    Surface surface = null;
+                    for (final Map<String, Object> propositionMap : retrievedPropositions) {
+                        final Proposition proposition = Proposition.fromEventData(propositionMap);
```

Comment:
1. Perform null check before invoking `proposition.getScope()` as `fromEventData(...)` can return null. 
2. The logic here doesn't accommodate the fact that if there are multiple propositions found for the same scope, they need to be accumulated in the list and then added to requestedPropositions map (against that scope).